kernel/arch/dreamcast/sound/snd_iface.c

```diff
@@ -27,6 +28,11 @@ static int initted = 0;
 extern uint8_t snd_stream_drv[];
 extern uint8_t snd_stream_drv_end[];
 
+/* The queue processing mutex for snd_sh4_to_aica_start and snd_sh4_to_aica_stop.
+   There are rare case, start stream + sfx control at the same time in separate threads,
+   so we just use fast spinlock for it. */
+static spinlock_t queue_proc_mutex = SPINLOCK_INITIALIZER;
```

Comment:
This is not a mutex, it is a spinlock. So name it accordingly.

But it sounds like your brain really wanted you to have a mutex, so why use a spinlock? It sounds like having a mutex would be a much better idea.

Reply:
I have some doubts here. It's just that this lock is needed extremely rarely and I don't want to pull the mutex for no reason.

Reply:
Although, better really make a mutex, I remembered other cases.